src/main/java/Timo/Timo.java

```diff
@@ -0,0 +1,584 @@
+    }
+
+    /**
+     * given Event task and size, prints out that Event task has been added, and prints out the total size of the array
+     * @param event
+     * @param size
+     */
+    public void printEvent(Event event, int size) {
+        System.out.println("----------------------------");
+        System.out.println("Got it. I've added this task:");
+        System.out.println(event.toString());
+        System.out.println("Now you have " + size + " tasks in the list.");
+        System.out.println("----------------------------");
+    }
+
+    /**
+     * given the task and the size, prints out the task that was deleted and the size of the array
+     * @param task
+     * @param size
+     */
+    public void printDelete(Task task, int size) {
+        System.out.println("----------------------------");
+        System.out.println("Got it. I've removed this task:");
+        System.out.println(task.toString());
+        System.out.println("Now you have " + size + " tasks in the list.");
+        System.out.println("----------------------------");
+    }
+
+    /**
+     * prints out the error
+     * @param e
+     */
+    public void printUnknownCommandError(TimoException e) {
+        System.out.println("----------------------------");
+        System.out.println(e);
+        System.out.println("----------------------------");
+    }
+
+}
+
+
+//Timo.Parser: deals with making sense of the commands
+//parser deal, then send commands to ui
+class Parser {
+    private UI ui;
+    private Storage storage;
+    private TaskList taskList;
+    public Parser(UI ui, Storage storage, TaskList taskList) {
+        this.ui = ui;
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Parses and executes a given command string.
+     * <p>
+     * The method interprets the command and performs the corresponding action, such as marking tasks,
+     * adding tasks (Todo, Deadline, Event), deleting tasks, printing the task list, or exiting the application.
+     * If the command is not recognized or lacks necessary arguments, a `TimoException` is thrown.
+     * </p>
+     *
+     * @param command The command string input by the user.
+     *
+     * @throws TimoException If the command is invalid or does not include the necessary arguments.
+     *
+     * @see TaskList
+     * @see Task
+     * @see Todo
+     * @see Deadline
+     * @see Event
+     * @see TimoException
+     */
+    public void parse(String command) throws TimoException {
+        switch (command) {
+            case "bye":
+                this.ui.bye();
+                this.storage.store(this.taskList.showList());
+                break;
+
+            case "list":
+                this.ui.printList(this.taskList);
+                break;
+
+            default:
+                String s = String.valueOf(command.charAt(command.length() - 1));
+                if (command.startsWith("mark")) {
```

Comment:
Any reason why mark, unmark, delete, todo, deadline, event and find are not cases of the switch block?

Reply:
You are right, I will go edit it!